worker/src/tests/commons.rs

```diff
@@ -132,3 +137,44 @@ pub fn get_nonce(api: &Api<sr25519::Pair>, who: &AccountId32) -> u32 {
         0
     }
 }
+
+#[cfg(test)]
+pub fn test_sidechain_block() -> SignedBlock {
+    use sp_core::{H256, Pair};
+
+
+    let signer_pair = sp_core::ed25519::Pair::from_string("//Alice", None).unwrap();
+    let author: AccountId32 = signer_pair.public().into();
+    let block_number: u64 = 0;
+    let parent_hash = H256::random();
+    let layer_one_head = H256::random();
+    let signed_top_hashes = vec![];
+    let encrypted_payload: Vec<u8> = vec![];
+    let shard = ShardIdentifier::default();
+
+    // when
+    let block = Block::construct_block(
+        author,
+        block_number,
+        parent_hash.clone(),
+        layer_one_head.clone(),
+        shard.clone(),
+        signed_top_hashes.clone(),
+        encrypted_payload.clone(),
+    );
+    block.sign(&signer_pair)
+}
```

Comment:
When it comes to constructing test data objects, I started relying quite a bit on the builder pattern. A builder will have a `build()` method and for each parameter you want to set, a `with_xyz()` method, that returns the builder itself and sets that parameter. Internally all parameters are options, so when building the object, we know which parameter was set by the client and which parameters need to have default values. 

I haven't used it yet in Rust, so I'll have to write an example first to better illustrate what I mean 😄

Reply:
I am also favoring that, and I have seen before it in rust. No need to illustrate 👍

btw. I would also suggest such a pattern in the enclave as a replacement for `init_io_handler()`

Reply:
In PolkaDex I've reworker the implementation of `init_io_handler()` by basically abstracting the RPC calls, see 
https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/0c66ae8bf005719398bf4b294108ef18d6695841/enclave/src/rpc/worker_api_direct.rs#L119-L139 in the `init_io_handler` and https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/0c66ae8bf005719398bf4b294108ef18d6695841/enclave/src/rpc/rpc_cancel_order.rs#L32-L91 for an example of an RPC call implementation

Reply:
Ah i see, this is a big improvement. This is basically the beginning of what I thought. We could modularize this a bit more e.g. let's say polkadex also has a sidechain rpc. What I'd like to do is:

```
let io_handler = IoHandlerBuilder::new()
   .with_order_api()
   .with_sidechain_api()
   .build() <- collect all the rpcs here in an rpc_methods function.
```

Reply:
Ah yes, using a builder pattern like this would make it very elegant and concise 👍